src/FakeAsyncClient.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Utils;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+use Http\Client\Exception;
+use Http\Client\Utils\Promise\FulfilledPromise;
+use Http\Client\Utils\Promise\RejectedPromise;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Transform a HttpClient into a HttpAsyncClient
```

Comment:
maybe 

Emulate an HttpAsyncClient on top of a HttpClient.

This client sends request synchronously but then uses the Promise system to allow a consistent codebase even with non-asynchronous capable clients.